Bottom: a02c9620191e079bc06312ca51e580cc5d1b3fcd
Top:    a9835af0e327ed185b532293742e6adf71e8c8ad
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2014-07-05 17:01:17 +0100

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index c402568..14590b5 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2460,7 +2460,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
